### PR TITLE
feat: introduce legacy sizing API and behavior

### LINKED ISSUE #17: Introduce sizing API on com.vaadin.legacy.ui.AbstractComponent level
<html>
<body>

Return| Method | How
-- | -- | --
float | getHeight()Gets the height of the object. | Custom - incompatible with `HasSize` which returns `String`
Sizeable.Unit | getHeightUnits()Gets the height property units.  | Custom - incompatible with `HasSize` which returns `Optional`
float | getWidth()Gets the width of the object. |  | Custom - incompatible with `HasSize` which returns `String`
Sizeable.Unit | getWidthUnits()Gets the width property units.  | Custom - incompatible with `HasSize` which returns `Optional`
void | setHeight(float height, Sizeable.Unit unit)Sets the height of the object. | Same as `HasSize`
void | setHeight(String height)Sets the height of the component using String presentation.  | Same as `HasSize`
void | setHeightFull()Sets the height to 100%. | Same as `HasSize`
void | setHeightUndefined()Clears any defined height. | Custom
void | setSizeFull()Sets the size to 100% x 100%. | Same as `HasSize`
void | setSizeUndefined()Clears any size settings. | Same as `HasSize`
void | setWidth(float width, Sizeable.Unit unit)Sets the width of the object. | Same as `HasSize`
void | setWidth(String width)Sets the width of the component using String presentation. | Same as `HasSize`
void | setWidthFull()Sets the width to 100%. | Same as `HasSize`
void | setWidthUndefined()Clears any defined width. | Custom

</body>
</html>

On framework the the API is in `com.vaadin.server.Sizeable` interface, which can be reintroduced together with the `Sizeable.Unit` enum:
```
CM Unit code representing centimeters.
EM Unit code representing the font-size of the relevant font.
EX Unit code representing the x-height of the relevant font.
INCH Unit code representing inches.
MM Unit code representing millimeters.
PERCENTAGE Unit code representing in percentage of the containing element defined by terminal.
PICAS Unit code representing picas (12 points).
PIXELS Unit code representing pixels.
POINTS Unit code representing points (1/72nd of an inch).
REM Unit code representing the font-size of the root font.
```

### PR DESCRIPTION
**Pull request by pleku**
Adds the legacy setWidth & setHeight APIs and the behavior. Similarly
to legacy framework, for relative sizes the CSS width/height is not
applied to the component in certain scenarios depending on the parent
and even sizing of the sibling components.

Closes #17

### CODE REVIEW

**Comment by pleku** [comment]
Blocked until #42 is done

File: vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponentContainer.java [hunk 1]
```diff
@@ -33,7 +36,7 @@
  * @author Vaadin Ltd
  */
 public abstract class AbstractComponentContainer extends AbstractComponent
-        implements ComponentContainer {
+        implements ComponentContainer, Iterable<Component> {
```
Comment (fhdhsni) [inline vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponentContainer.java:39]: `Iterable<Component>` can be moved to `HasComponents`

File: vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponentContainer.java [hunk 2]
```diff
@@ -76,4 +85,46 @@ public void removeAllComponents() {
...
+    /**
+     * Gets the component container iterator for going through all the
+     * components in the container.
+     *
+     * @return the Iterator of the components inside the container.
+     */
+    @Override
+    public Iterator<Component> iterator() {
+        return Collections.unmodifiableCollection(
+                getChildren().collect(Collectors.toList())).iterator();
```
Comment (fhdhsni) [inline vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponentContainer.java:104]: can't we use `getChildren.iterator();`? Streams also throw `UnsupportedOperationException` when you can `remove` on their iterators

File: vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/server/ComponentSizeValidator.java [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.legacy.server;
+
+import java.io.Serializable;
+import java.util.Iterator;
+import java.util.logging.Logger;
```
Comment (fhdhsni) [inline vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/server/ComponentSizeValidator.java:20]: unused import

File: vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/server/ComponentSizeValidator.java [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+    }
+
+    /**
+     * Internal framework sizing magic method that has never had javadocs. The
+     * purpose is to block users from setting relative (%) size child components
+     * when inside certain legacy components.
+     * 
+     * @param component
+     *            the flow or legacy component
+     * @return whether
```
Comment (fhdhsni) [inline vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/server/ComponentSizeValidator.java:49]: whether parent can define height or not...

File: vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponentContainer.java [hunk 5]
```diff
@@ -15,11 +15,14 @@
  * #L%
  */
 
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
```
Comment (fhdhsni) [inline vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponentContainer.java:21]: nit: unused import

Comment (pleku) [reply vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponentContainer.java:21]: Done

File: vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/server/ComponentSizeValidator.java (hunk 4 above)
Comment (pleku) [reply vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/server/ComponentSizeValidator.java:49]: Done

File: vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponentContainer.java (hunk 2 above)
Comment (pleku) [reply vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponentContainer.java:104]: Correct, good catch.

File: vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/server/ComponentSizeValidator.java (hunk 3 above)
Comment (pleku) [reply vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/server/ComponentSizeValidator.java:20]: Done

File: vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponentContainer.java (hunk 1 above)
Comment (pleku) [reply vaadin-legacy-components-flow/src/main/java/com/vaadin/legacy/ui/AbstractComponentContainer.java:39]: Done

**Review by fhdhsni — APPROVED** [review]

**Review by fhdhsni — APPROVED** [review]

**Review by DiegoCardoso — APPROVED** [review]

**Pull request merged**